std-bits/table/src/main/java/org/enso/table/aggregations/MinOrMax.java

```diff
@@ -0,0 +1,67 @@
+package org.enso.table.aggregations;
+
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.table.Column;
+import org.enso.table.data.table.problems.InvalidAggregation;
+
+import java.util.List;
+
+/***
+ * Aggregate Column finding the minimum or maximum entry in a group.
+ */
+public class MinOrMax extends AggregateColumn{
+  private final Storage storage;
+  private final int minOrMax;
+
+  public MinOrMax(String name, Column column, int minOrMax) {
+    super(name, Storage.Type.OBJECT);
+    this.storage = column.getStorage();
+    this.minOrMax = minOrMax;
+  }
+
+  @Override
+  public Object aggregate(List<Integer> indexes) {
+    Object current = null;
+    for (int row: indexes) {
+      Object value = storage.getItemBoxed(row);
+      if (value != null) {
+        try {
+          if (current == null || Compare(current, value) == minOrMax) {
+            current = value;
+          }
+        } catch (ClassCastException e) {
+          this.addProblem(new InvalidAggregation(this.getName(), row, "Cannot compare values."));
+          return null;
+        }
+      }
+    }
+    return current;
+  }
+
+  private static int Compare(Object current, Object value) {
+    if (current instanceof String && value instanceof String) {
+      return ((String)value).compareTo((String)current);
+    }
```

Comment:
This will not be aligned with our String normalization.

I.e. for strings `'ś'` and `'s\u{301}'` it will treat them as unequal even though in-Enso they are equal. I think we should strive to be consistent.
Not saying this must necessarily be done in this PR, but I think we should at least open a ticket to fix this.

Reply:
I'm wondering if we could just make `std-table` depend on `std-base` so we can easily re-use all the Unicode handling helpers that we define there. I think this should be the right approach here (as we don't want to re-implement them, as this will easily get out of sync).

Reply:
will add a ticket

Reply:
https://www.pivotaltracker.com/story/show/181754646